### PR TITLE
ARROW-15133: [CI] Remove util_checkout.sh and util_cleanup.sh scripts

### PR DESCRIPTION
**Pull request by kszucs**
- `ci/scripts/util_checkout.sh` was used to checkout submodules because `actions/checkout@v2` has removed support for that, but they have restored it since.
- `ci/scripts/util_cleanup.sh` was used to free up disk space on github actions runners, because at that time it was limited to 7GB, from a recent run it looks like the linux runners now have 32GB free space so we can try to disable the cleanup step sparing almost a minute of build time

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit -g nightly

**Comment by kszucs** [comment]
@github-actions crossbow submit -g nightly

**Comment by kszucs** [comment]
> Failed to render template `python-wheels/github.linux.amd64.yml` with UndefinedError: 'archery.crossbow.core.Target object' has no attribute 'github_repo'
> The Archery job run can be found at: https://github.com/apache/arrow/actions/runs/1587734449```

This is going to work after the PR gets merged.

Submitted the tasks manually: https://github.com/ursacomputing/crossbow/branches/all?query=build-579

**Review by pitrou — APPROVED** [review]
Nice cleanup!

File: .github/workflows/cpp.yml [hunk 1]
```diff
@@ -70,10 +70,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - name: Fetch Submodules and Tags
-        run: ci/scripts/util_checkout.sh
-      - name: Free Up Disk Space
-        run: ci/scripts/util_cleanup.sh
+          submodules: recursive
```
Comment (jonkeane) [inline .github/workflows/cpp.yml:73]: I noticed that some checkouts have `submodules: recursive` and others don't — is there a pattern here that I'm missing?

**Review by jonkeane — APPROVED** [review]
This looks great, thanks for all this! One small question (for my edification more than anything)

File: .github/workflows/cpp.yml (hunk 1 above)
Comment (kszucs) [reply .github/workflows/cpp.yml:73]: Correct, we don't run flight/parquet/integration tests everywhere to can spare some additional time by omitting the submodule checkouts.

File: .github/workflows/js.yml [hunk 2]
```diff
@@ -111,7 +104,7 @@ jobs:
       - name: Checkout Arrow
         uses: actions/checkout@v1
```
Comment (kou) [inline .github/workflows/js.yml:105]: `v1` -> `v2`

Comment (jonkeane) [reply .github/workflows/js.yml:105]: Thanks for catching this, I've created https://issues.apache.org/jira/browse/ARROW-15140 and I'll follow up with that shortly.

Comment (jonkeane) [reply .github/workflows/js.yml:105]: Oh, ha it was _just that one_ too! Anyway: https://github.com/apache/arrow/pull/11983/files

**Pull request closed without merging**